baseplate/frameworks/pyramid/__init__.py

```diff
@@ -83,8 +139,43 @@ def baseplate_tween(request: Request) -> Response:
         else:
             if request.span:
                 request.span.set_tag("http.status_code", response.status_code)
+                content_length = response.content_length
                 response.app_iter = SpanFinishingAppIterWrapper(request.span, response.app_iter)
+                response.content_length = content_length
         finally:
+            http_endpoint = endpoint
+            http_method = request.method.lower()
+            http_response_code = ""
+
+            if sys.exc_info() == (None, None, None):
+                http_success = (
+                    getHTTPSuccessLabel(int(response.status_code)) if response else "false"
+                )
+                http_response_code = response.status_code if response else ""
+            else:
+                http_success = "false"
+
+            histogram_labels = {
+                "http_method": http_method,
+                "http_endpoint": http_endpoint,
+                "http_success": http_success,
+            }
+            ACTIVE_REQUESTS.labels(http_method=http_method, http_endpoint=http_endpoint).dec()
+            REQUEST_LATENCY.labels(**histogram_labels).observe(time.perf_counter() - time_started)
+            REQUESTS_TOTAL.labels(
+                **{
+                    **histogram_labels,
+                    "http_response_code": http_response_code,
+                }
+            ).inc()
+            REQUEST_SIZE.labels(**histogram_labels).observe(request.content_length or 0)
+            if response:
+                try:
+                    if response.content_length is not None:
+                        RESPONSE_SIZE.labels(**histogram_labels).observe(response.content_length)
+                except Exception:
```

Comment:
I believe this is the handling of the case where it errors and content_length is not set. This is fine to me, but a comment would be nice

Reply:
It's really just guarding against the case content_length is not a number... I can add a comment but I'm not 100% what I would say outside of "we can't observe None / not an integer so we just don't"